monai/losses/perceptual.py

```diff
@@ -0,0 +1,371 @@
+    """
+    Perceptual loss using features from pretrained deep neural networks trained. The function supports networks
+    pretrained on: ImageNet that use the LPIPS approach from Zhang, et al. "The unreasonable effectiveness of deep
+    features as a perceptual metric." https://arxiv.org/abs/1801.03924 ; RadImagenet from Mei, et al. "RadImageNet: An
+    Open Radiologic Deep Learning Research Dataset for Effective Transfer Learning"
+    https://pubs.rsna.org/doi/full/10.1148/ryai.210315 ; MedicalNet from Chen et al. "Med3D: Transfer Learning for
+    3D Medical Image Analysis" https://arxiv.org/abs/1904.00625 ;
+    and ResNet50 from Torchvision: https://pytorch.org/vision/main/models/generated/torchvision.models.resnet50.html .
+
+    The fake 3D implementation is based on a 2.5D approach where we calculate the 2D perceptual on slices from the
+    three axis.
+
+    Args:
+        spatial_dims: number of spatial dimensions.
+        network_type: {``"alex"``, ``"vgg"``, ``"squeeze"``, ``"radimagenet_resnet50"``,
+        ``"medicalnet_resnet10_23datasets"``, ``"medicalnet_resnet50_23datasets"``, ``"resnet50"``}
+            Specifies the network architecture to use. Defaults to ``"alex"``.
+        is_fake_3d: if True use 2.5D approach for a 3D perceptual loss.
+        fake_3d_ratio: ratio of how many slices per axis are used in the 2.5D approach.
+        cache_dir: path to cache directory to save the pretrained network weights.
+        pretrained: whether to load pretrained weights. This argument only works when using networks from
+            LIPIS or Torchvision. Defaults to ``"True"``.
+        pretrained_path: if `pretrained` is `True`, users can specify a weights file to be loaded
+            via using this argument. This argument only works when ``"network_type"`` is "resnet50".
+            Defaults to `None`.
+        pretrained_state_dict_key: if `pretrained_path` is not `None`, this argument is used to
+            extract the expected state dict. This argument only works when ``"network_type"`` is "resnet50".
+            Defaults to `None`.
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int,
+        network_type: str = "alex",
+        is_fake_3d: bool = True,
+        fake_3d_ratio: float = 0.5,
+        cache_dir: str | None = None,
+        pretrained: bool = True,
+        pretrained_path: str | None = None,
+        pretrained_state_dict_key: str | None = None,
+    ):
+        super().__init__()
+
+        if spatial_dims not in [2, 3]:
+            raise NotImplementedError("Perceptual loss is implemented only in 2D and 3D.")
+
+        if (spatial_dims == 2 or is_fake_3d) and "medicalnet_" in network_type:
+            raise ValueError(
+                "MedicalNet networks are only compatible with ``spatial_dims=3``."
+                "Argument is_fake_3d must be set to False."
+            )
+
+        if cache_dir:
+            torch.hub.set_dir(cache_dir)
```

Comment:
this will potentially change the caching location of all the default bundles defined here: https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/monai/bundle/scripts.py#L212-L214

perhaps it's helpful to have a warning here